lib/wallet/txdb.js

```diff
@@ -690,8 +690,22 @@ class TXDB {
 
       details.setOutput(i, path);
 
-      const credit = await this.getCredit(hash, i);
-      assert(credit);
+      let credit = await this.getCredit(hash, i);
+
+      if (!credit) {
+        // This credit didn't belong to us the first time we
+        // saw the transaction (before confirmation or rescan)
+        this.logger.warning(
```

Comment:
Necessary for this to be a warning?

Reply:
What level do you recommend? I chose warning because it is an edge case, and would be an expected event. I think it should stand out in the log to indicate that there might be a bigger issue for the user (too-small lookahead, for example)

Reply:
I don't think it indicates any other issue. So there may not need to be a log at all.

Reply:
It could be worth logging "add transaction again with additional output" to some effect. This is a case where `confirm` isn't confirming but also inserting.